### ISSUE TITLE
Reproducing Paper Accuracy for OverLoCK

### ISSUE THREAD

**Issue by IONSliu** [LMMMEng/OverLoCK#27]
hi,

Thank you for your excellent work! I am very interested in your code.I’m trying to reproduce the accuracy results from the paper, where you mention adhering to the same experimental setting described in DeiT [63] for a fair comparison. However, I noticed that the code sets the batch size to 256 per GPU and uses \( args.lr = (args.batch_size 256 * args.world_size 8/ 1024) * 1e-3 \),which seems inconsistent with the paper’s learning rate formula of 0.0005 *s batchsize(1024) / 512 [63].

I’m planning to run experiments on 8 NVIDIA 4090 GPUs, but due to memory constraints, the maximum batch size per GPU is 128, resulting in a total batch size of 128*8 = 1024 ), matching the paper’s [63]setting。Could you please advise on which parameters I should adjust, such as the learning rate scaling formula, to ensure the best possible reproduction of the 82.7 top-1 accuracy for the XT version on ImageNet-1K as reported in your paper?

Thank you for your help!

**Comment by IONSliu** [comment]
At the same time, could the training logs be made public to ensure that researchers can reproduce the results?

**Comment by LMMMEng** [comment]
Hello, the scaling strategy for learning rate (LR) is the same as DEIT, where the LR is adjusted based on the batch size (BS). For example, a BS of 1024 corresponds to an LR of 0.001, while a BS of 2048 corresponds to an LR of 0.002. We followed the same training strategy as DEIT, including the number of epochs, data augmentation, and optimizer, but with one minor difference: we used a larger BS to improve training efficiency, as training on ImageNet is computationally expensive. This approach is inspired by previous works such as BiFormer, MetaFormer, and UniRepLKNet.

I'm not sure if reducing the BS from 2048 to 1024 would affect accuracy. Theoretically, our model, which includes many batch normalization (BN) layers, should benefit from a larger BS. If you want to reproduce the results, I highly suggest using the provided configuration. However, if you're running low on GPU memory, you can consider using gradient checkpointing. For example, you can add the following command to training script:

```--grad-checkpoint --ckpt-stg 2 0 0 0```

This enables gradient checkpointing and sets the number of blocks for each stage to 2, 0, 0, and 0, respectively. You can increase these values if you're still running low on memory, but keep in mind that this may slow down training. 

The training log will be released later, as I had to resume training multiple times on different devices, and organizing the logs will take some time. I'm currently busy with another project, so thank you for your patience.

**Comment by IONSliu** [comment]
Thank you very much. I will try.


Meng Lou ***@***.***>于2025年5月25日 周日16:21写道：
